diff --git a/scripts/Makefile.spl b/scripts/Makefile.spl
--- a/scripts/Makefile.spl
+++ b/scripts/Makefile.spl
@@ -437,6 +437,12 @@
 $(obj)/u-boot-spl-dtb.hex: $(obj)/u-boot-spl-dtb.bin FORCE
 	$(call if_changed,objcopy)

+ifeq ($(CONFIG_SPL_IMAGE_TYPE_SUNXI_TOC0),y)
+$(obj)/sunxi-spl.bin: root_key.pem
+root_key.pem:
+	openssl genrsa -out $@ 2048
+endif
+
 $(obj)/sunxi-spl.bin: $(obj)/$(SPL_BIN).bin FORCE
 	$(call if_changed,mkimage)
